core/src/main/scala/chisel3/experimental/hierarchy/core/Lookupable.scala

```diff
@@ -342,10 +342,11 @@ object Lookupable {
       def instanceLookup[A](that: A => B, instance: Instance[A]): C = {
         val ret = that(instance.proto)
 
+        // As Property ports are not yet Lookupable, they are skipped here.
         def getIoMap(hierarchy: Hierarchy[_]): Option[Map[Data, Data]] = {
           hierarchy.underlying match {
             case Clone(x: ModuleClone[_]) => Some(x.ioMap)
-            case Proto(x: BaseModule) => Some(x.getChiselPorts.map { case (_, data) => data -> data }.toMap)
+            case Proto(x: BaseModule) => Some(x.getChiselPorts.map { case (_, data: Data) => data -> data }.toMap)
```

Comment:
I'm filtering to only Data here. I guess when we want non-Data (i.e., Properties) to work here, we'd make a new Lookupable typeclass instance for Properties?

Reply:
@jackkoenig was arguing we'd want to up-level this lookupable to be able to look for the superclass `(IsBindable)`. This makes it simpler to have e.g. sequence, though, I'd argue you can also just have a sequence of `Lookupable` or you'd have the same problem anyway (not sure why a sequence of Data and Property is more special than say a sequence of Data and some other IsInstantiable...)

Reply:
Yeah I think this is pretty much the main question I have on this PR. If I just do this (pretend non-Data ports don't exist), I can make headway on using properties in ports without having to make them work with Lookupable and all the D/I stuff. If I do pull on the thread of making this look for IsBindable, the core assumption that Lookupable works for Data needs to be generalized to IsBindable in a lot of places.

Reply:
I think this is fine for now, we can work on Properties and perhaps fully generic IsBindables in D/I later.